internal.nix

```diff
@@ -300,9 +304,21 @@ rec {
       assert (builtins.typeOf preInstallLinks != "set") ->
         throw "`preInstallLinks` must be an attributeset of attributesets";
       let
-        cleanArgs = builtins.removeAttrs args [ "src" "packageJson" "packageLockJson" "buildInputs" "nativeBuildInputs" "nodejs" "preBuild" "postBuild" "preInstallLinks" "githubSourceHashMap" ];
+        cleanArgs = builtins.removeAttrs args [ "src" "packageJson" "packageLockJson" "buildInputs" "nativeBuildInputs" "nodejs" "preBuild" "postBuild" "preInstallLinks" "githubSourceHashMap" "pname" "version" ];
         lockfile = readLockfile packageLockJson;
 
+        # allow to set package-name and version, if name or version are missing in package-lock.json
+        pname =
+          if (args ? pname && args.pname != "") then makeValidDrvName args.pname
+          else if (lockfile ? name && lockfile.name != "") then lockfile.name
+          else ""; # pname is required, throw later
```

Comment:
Can we use `null` here instead?  It doesn't make much of a difference but I kinda like defaulting to a specific value for this that isn't a string (and thus is harder to accidentially use wrong).